ee/clickhouse/queries/session_recordings/clickhouse_session_recording_list.py

```diff
@@ -40,16 +40,17 @@ class ClickhouseSessionRecordingList(ClickhouseEventQuery):
 
     _event_and_recording_match_conditions_clause = """
         (
-            -- If there is a window_id on the recording, then it is newer data and we can match
-            -- the recording and events on session_id
+            -- If there is a session_id on the event, then it is from posthog-js
+            -- and we should use the session_id to match the recording
             (
-                notEmpty(session_recordings.window_id) AND
+                notEmpty(events.session_id) AND
                 events.session_id == session_recordings.session_id
             ) OR
-            -- If there's no window_id on the recording, then it is older data and we should match
-            -- events and recordings on timestamps
+            -- If there's no session_id on the event, then it is either older data or
+            -- from a posthog-client that doesn't support session (e.g. backend client)
+            -- and we should match on timestamp
             (
-                empty(session_recordings.window_id) AND
+                empty(events.session_id) AND
```

Comment:
In the first condition will an event that has a `session_id` also always have a `window_id`? It seems that we're changing the behavior here to check for `session_id` instead of `window_id` which could break older posthog-js clients. I feel like I'm missing some context here though, so maybe I just need some clarification!

Reply:
Good question! The old check was for backward compatibility, but we added `window_id` to `session_recording_events` at the same time that we added `session_id` and `window_id` to events (in this [PR](https://github.com/PostHog/posthog-js/pull/326)), so it shouldn't matter which we're checking for. 

So now, this check is serving the same purpose as the backward compatibility check, but it's also handling the on-going case for server-side events.

Reply:
Thanks for the context! Good to know that this check is the same either way.